lisa/tests/base.py

```diff
@@ -475,8 +475,11 @@ def trace_window(self, trace):
 
         return (rta_start, rta_stop)
 
+    # Use LRU cache instead of memoized, to avoid caching the trace forever, in
+    # case the thread is manipulating a large number of TestBundles without
+    # deleting them.
     @property
-    @memoized
+    @functools.lru_cache(maxsize=50, typed=True)
```

Comment:
We could probably reduce that further to something like 20/30. You rarely re-execute things on more than that.